documentation/recovering_lost_account_due_to_email_provider_change.md

```diff
@@ -0,0 +1,157 @@
+# Recovering a lost KBase account due to an email provider change
+
+## Document purpose
+
+This document describes the steps that were taken to restore access to a user's KBase account
+after their email provider changed.
+
+The document, firstly, will hopefully help resolve similar situations in the future, and secondly,
+recommends improvements to the recovery process.
+
+## Background
+
+A user had a government supplied email address (e.g. like lbl.gov) that was provided by Google.
+They had a KBase account linked to two identity suppliers - the first was Google, using the 
+government address, and the second Globus, using the Google address:
+
+```
+          KBase
+          |   \
+          | Globus
+          |   /
+         Google
+           |
+         agency.gov
+```
+
+The government agency switched email suppliers away from Google, and so the user could no
+longer log into Google using that email address. This meant they could no longer log into Google,
+Globus, or KBase.
+
+## Resolution
+
+The user's ultimate email address (e.g. the *.gov address) was the same for the KBase, Google,
+and Globus accounts based on examination of the user's record in the auth database, and so it
+was decided to treat the address as trusted. **It is not necessarily true** that an email
+address in the database can be trusted - KBase email addresses are unverified, and Globus
+provides no guarantees about the email records in its database.
+
+The user had an ORCiD that was unused in KBase and provided it to KBase personnel via the
+trusted email address, thus making the ORCiD trusted.
+
+The KBase identity ID for the user's ORCiD account was calculated via a small piece of code:
+
+```
+package us.kbase.auth2;
+
+import us.kbase.auth2.lib.identity.RemoteIdentityID;
+
+public class TempHelper {
+
+	public static void main(String[] args) {
+		final String identityProvider = args[0];
+		final String remoteID = args[1];
+		final RemoteIdentityID rid = new RemoteIdentityID(identityProvider, remoteID);
+		System.out.println(rid);
+		System.out.println("Kbase identity ID " + rid.getID());
+	}
+}
+```
+
+When run with `OrcID` and the user's OrcID as the input, the program produces:
+
+```
+RemoteIdentityID [provider=OrcID, id=<users's OrcID>]
+Kbase identity ID <KBase identity ID>
+```
+
```

Comment:
I verified the user's entry in mongo with to make sure there was only 1 match for the username:
kbrs0:PRIMARY> db.users.findOne({user: "epbarnhart"})
{
	"_id" : ObjectId("5991c1f55908010007ca4a11"),
	"user" : "epbarnhart",
	"lcl" : false,
	"email" : "epbarnhart@usgs.gov",
	"display" : "Elliott Barnhart",
	"dispcan" : [
		"elliott",
		"barnhart"
	],
	"roles" : [ ],
	"custrls" : [ ],
	"idents" : [
		{
			"id" : "3b6cc265bb0f216647e6525bbe2e3ba4",
			"prov" : "Globus",
			"prov_id" : "e36f7bb8-5c95-483c-bba8-a098d1228ff1",
			"uid" : "epbarnhart@usgs.gov@accounts.google.com",
			"fullname" : "Elliott Barnhart",
			"email" : "epbarnhart@usgs.gov"
		},
		{
			"id" : "4d4aac54b9625930e3454192cbf61a93",
			"prov" : "OrcID",
			"prov_id" : "0000-0002-8788-8393",
			"uid" : "0000-0002-8788-8393",
			"fullname" : "Elliott Barnhart",
			"email" : "epbarnhart@usgs.gov"
		}
	],
	"policyids" : [
		{
			"id" : "data-policy.1",
			"agreed" : ISODate("2017-08-14T15:29:57.533Z")
		},
		{
			"id" : "kbase-user.1",
			"agreed" : ISODate("2017-08-14T15:29:57.533Z")
		}
	],
	"create" : ISODate("2017-08-14T15:29:57.533Z"),
	"login" : ISODate("2020-05-20T22:10:07.635Z"),
	"dsbleadmin" : null,
	"dsbledate" : null,
	"dsblereas" : null
}

I would include that in the doc as a "best practice".